src/sentry/integrations/utils/code_mapping.py

```diff
@@ -276,16 +276,29 @@ def _generate_code_mapping_from_tree(
             for src_path in repo_tree.files
             if self._potential_match(src_path, frame_filename)
         ]
+        # Allow for empty for ./ as stack root if the following cases apply
+        stacktrace_root = f"{frame_filename.root}/"  # sentry
```

Comment:
Move all of this code up to the return into a function called `_normalized_stack_and_source_roots`. You can then write unit tests for it.

```python
stacktrace_root, source_path = _normalized_stack_and_source_roots(f"{frame_filename.root}/", self._get_code_mapping_source_path(matched_files[0], frame_filename))
```